### PR TITLE
INT-3860 split the jenkinsfiles for a snapshot and a release build

### PR DESCRIPTION
**Pull request by jflinchbaugh**
We're instating a master snapshot build for docker-nexus3. Jenkinsfile will be the default build that is used for snapshot builds/tests/scans of features and master. Jenkinsfile-Release will be used by the existing docker-nexus3 build in jenkins to do a release. That existing job will not be renamed, but it will be rewired to run Jenkins-Release, which is an unchanged copy of the old Jenkinsfile.

JIRA: https://issues.sonatype.org/browse/INT-3860
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/docker-nexus3-feature/job/INT-3860-snapshot-build/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -7,18 +7,8 @@
...
 
-properties([
-  parameters([
-    string(defaultValue: '', description: 'New Nexus Repository Manager Version', name: 'nexus_repository_manager_version'),
-    string(defaultValue: '', description: 'New Nexus Repository Manager Version Sha256', name: 'nexus_repository_manager_version_sha'),
-    string(defaultValue: '', description: 'New Nexus Repository Manager Cookbook Version', name: 'nexus_repository_manager_cookbook_version'),
-    booleanParam(defaultValue: false, description: 'Skip Pushing of Docker Image and Tags', name: 'skip_push'),
-    booleanParam(defaultValue: false, description: 'Only update the latest tag', name: 'update_latest_only')
-  ])
-])
```
Comment (jflinchbaugh) [inline Jenkinsfile:18]: Everything removed from this file is preserved in the new Jenkinsfile-Release file. This one will only be for slimmer snapshot builds of feature and master branches.

Comment (eduard-tita) [reply Jenkinsfile:18]: Does the `Jenkinsfile-Release` file contain everything that used to be in the original `Jenkinsfile`? It's hard to tell.

Comment (mpiggott) [reply Jenkinsfile:18]: I'm ambivalent about it aside from suggesting it as a way for `git` to understand what happened. :)

I believe `git` only recognizes moves heuristically so unfortunately using `git mv` first won't help once we squash.

Comment (jflinchbaugh) [reply Jenkinsfile:18]: I tried it locally, `git mv Jenkinsfile Jenkinsfile-Release` and then `cp Jenkinsfile-Release Jenkinsfile` back, but once the files were added, git lost the trail:
```
jflinchbaugh@Johns-MacBook-Pro:~/workspace/docker-nexus3
$ g s
On branch mv-test
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	renamed:    Jenkinsfile -> Jenkinsfile-Release

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	Jenkinsfile

jflinchbaugh@Johns-MacBook-Pro:~/workspace/docker-nexus3
$ g add .
jflinchbaugh@Johns-MacBook-Pro:~/workspace/docker-nexus3
$ g s
On branch mv-test
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   Jenkinsfile
	new file:   Jenkinsfile-Release
```

File: Jenkinsfile-Release [hunk 2]
```diff
@@ -0,0 +1,249 @@
+/*
```
Comment (jflinchbaugh) [inline Jenkinsfile-Release:1]: this whole file is a copy of the original Jenkinsfile that used to do the release. It retains the guards that still allow it to only release from "master". I didn't see the need to remove those checks yet, but I could if reviewers insist.

**Review by eduard-tita — APPROVED** [review]
+1  LGTM

**Review by hectorhuol — APPROVED** [review]
+1

**Pull request merged**